manim/utils/decorators.py

```diff
@@ -0,0 +1,454 @@
+        # WARNING  The parameter runtime_in_ms of function foo has been deprecated and may be deleted in a future version.
+        # returns {"run_time": 0.5}
+
+    Redirecting multiple parameter values to one::
+
+        @deprecated_params(redirections=[
+            lambda buff_x=1, buff_y=1: {"buff": (buff_x, buff_y)}
+        ])
+        def foo(**kwargs):
+            return kwargs
+
+        foo(buff_x=2)
+        # WARNING  The parameter buff_x of function foo has been deprecated and may be deleted in a future version.
+        # returns {"buff": (2, 1)}
+
+    Redirect one parameter to multiple::
+
+        @deprecated_params(redirections=[
+            lambda buff=1: {"buff_x": buff[0], "buff_y": buff[1]} if isinstance(buff, tuple)
+                    else {"buff_x": buff,    "buff_y": buff}
+        ])
+        def foo(**kwargs):
+            return kwargs
+
+        foo(buff=0)
+        foo(buff=(1,2))
+        # WARNING  The parameter buff of function foo has been deprecated and may be deleted in a future version.
+        # returns {"buff_x": 0, buff_y: 0}
+
+        # WARNING  The parameter buff of function foo has been deprecated and may be deleted in a future version.
+        # returns {"buff_x": 1, buff_y: 2}
+
+
+    """
+    # Check if decorator is used without parenthesis
+    if callable(params):
+        raise ValueError("deprecate_parameters requires arguments to be specified.")
+
+    # Construct params list
+    params = re.split("[,\s]+", params) if isinstance(params, str) else list(params)
+
+    # Add params which are only implicitly given via redirections
+    for redirector in redirections:
+        if isinstance(redirector, tuple):
+            params.append(redirector[0])
+        else:
+            params.extend(redirector.__code__.co_varnames)
+    params = list(set(params))
+
+    # Make sure params only contains valid identifiers
+    identifier = re.compile(r"^[^\d\W]\w*\Z", re.UNICODE)
+    if not all(re.match(identifier, param) for param in params):
+        raise ValueError("Given parameter values are invalid.")
+
+    redirections = list(redirections)
+
+    def warning_msg(func: Callable, used: List[str]):
+        """Generate the deprecation warning message.
+
+        Parameters
+        ----------
+        func
+            The callable with deprecated parameters.
+        used
+            The list of depecated parameters used in a call.
+
+        Returns
+        -------
+        str
+            The deprecation message.
+        """
+        what, name = __get_callable_info(func)
+        plural = len(used) > 1
+        parameter_s = "s" if plural else ""
+        used_ = ", ".join(used[:-1]) + " and " + used[-1] if plural else used[0]
+        has_have_been = "have been" if plural else "has been"
+        deprecated = __deprecation_text_component(since, until, message)
+        return f"The parameter{parameter_s} {used_} of {what} {name} {has_have_been} {deprecated}"
```

Comment:
did you miss a space?
```suggestion
        return f"The parameter {parameter_s} {used_} of {what} {name} {has_have_been} {deprecated}"
```

Reply:
That's for plurality, so if there is more than one parameter it'll output `parameters`.

Reply:
didn't notice that :smile: